### PR TITLE
Nav 51 engine integration

### PR DESCRIPTION
**Pull request by tomeksabala**

### CODE REVIEW

File: navigator_api/api/main.py [hunk 1]
```diff
@@ -93,79 +96,55 @@ def workflow_state(dataset_id):
...
-            "dddeeefff",
-            "asdfasdfa",
-            "aaabbbbcc",
-            "dddeeefff",
-            "asdfasdfa"
-        ],
+        "progress": engine_decision["progress"]["progress"],
+        "milestones": engine_decision["progress"]["milestones"],
+        "milestoneListFullyResolved": engine_decision["progress"]["milestoneListFullyResolved"],
+        "taskBreadcrumps": engine_decision["actions"],
```
Comment (jonathansberry) [inline navigator_api/api/main.py]: Typo I think? "breadcrumps" > "breadcrumbs"

File: navigator_api/api/main.py [hunk 2]
```diff
@@ -93,79 +96,55 @@ def workflow_state(dataset_id):
...
-            ]
-        }
+        "skipped": is_task_skipped(dataset_id, task_id),
+        "details": _mock_task_details(task_details)
     })
 
 
+def _mock_task_details(task_details):
+    mock = {
+        "milestoneId": "6",
```
Comment (jonathansberry) [inline navigator_api/api/main.py:136]: We should talk about why you are adding `milestoneId` in here.

File: navigator_api/api/main.py [hunk 3]
```diff
@@ -93,79 +96,55 @@ def workflow_state(dataset_id):
...
-                "displayHtml": "<p><strong>Lorem Ipsum</strong> is simply dummy <br /> "
-                               "text of the printing and typesetting industry.</p>",
-                "skippable": True,
-                "helpUrls": [
-                    {"label": "Naomi help docs", "url": "http://example"},
-                    {"label": "Spectrum documentation", "url": "http://example"}
-                ]
-            }
+            "id": engine_decision["decision"]["id"],
+            "skipped": is_task_skipped(dataset_id, engine_decision["decision"]["id"]),
```
Comment (jonathansberry) [inline navigator_api/api/main.py:108]: I've been thinking... We need to add `removeSkips: [<node_id>]` to the engine response (or some similar solution) if the engine is recommending that you should remove some skipped steps in order to proceed. 

So if the engine recommends removing any skips, you should update the user state accordingly in the api layer.   This is essential for the skip logic to work.

**Review by jonathansberry — CHANGES_REQUESTED** [review]
Some thoughts plus a typo...

File: navigator_api/api/main.py (hunk 1 above)
Comment (tomeksabala) [reply navigator_api/api/main.py]: the spellings is correct I think. It's intentionally called "taskBreadcrumps" according to ui & api design.

File: navigator_api/api/main.py (hunk 2 above)
Comment (tomeksabala) [reply navigator_api/api/main.py:136]: the UI is displaying the current milestone of a browsed task.

File: navigator_api/api/main.py (hunk 3 above)
Comment (tomeksabala) [reply navigator_api/api/main.py:108]: Yeah, that sounds like a separate, longer discussion.

**Pull request merged**